### PR TITLE
v2.0.0 rc.5

### PR DESCRIPTION
**Pull request by Mullefa**
This fixes a bug where a mutable request builder was used by separate requests. This meant that the query string of a request would include the query string of all prior requests!

### CODE REVIEW

File: src/main/scala/com/gu/acquisition/services/DefaultOphanService.scala [hunk 1]
```diff
@@ -32,9 +31,13 @@ class DefaultOphanService(val endpoint: HttpUrl)(implicit client: OkHttpClient)
     import io.circe.syntax._
     import submission._
 
-    val url = additionalEndpointBuilder
-      .addQueryParameter("viewId", ophanIds.pageviewId)
-      .addQueryParameter("acquisition" , acquisition.asJson.noSpaces)
+    val url = endpoint.newBuilder()
+      .addPathSegment("a.gif")
+      // Checked against a local Ophan instance that the default Ok HTTP encoding was ok,
+      // however, until Ophan is upgraded, encode reserved characters as specified by RFC 3986 to be safe.
```
Comment (joelochlann) [inline src/main/scala/com/gu/acquisition/services/DefaultOphanService.scala]: It's actually `RFC 2396` - the older one is the stricter one

Comment (desbo) [reply src/main/scala/com/gu/acquisition/services/DefaultOphanService.scala]: I don't think we should introduce encodings in line with 2396, as Ophan definitely accepts requests with curly braces:

```
± curl -vg https://ophan.theguardian.com/a.gif\?viewId\=pageviewId\&acquisition\=\{%22product%22:%22CONTRIBUTION%22,%22paymentFrequency%22:%22ONE_OFF%22,%22currency%22:%22GBP%22,%22amount%22:20.0,%22paymentProvider%22:%22STRIPE%22,%22campaignCode%22:\[%22FAKE_ACQUISITION_EVENT%22\],%22abTests%22:\{%22tests%22:\[\{%22name%22:%22test_name%22,%22variant%22:%22variant_name%22\}\]\},%22countryCode%22:%22US%22\}

> GET /a.gif?viewId=pageviewId&acquisition={%22product%22:%22CONTRIBUTION%22,%22paymentFrequency%22:%22ONE_OFF%22,%22currency%22:%22GBP%22,%22amount%22:20.0,%22paymentProvider%22:%22STRIPE%22,%22campaignCode%22:[%22FAKE_ACQUISITION_EVENT%22],%22abTests%22:{%22tests%22:[{%22name%22:%22test_name%22,%22variant%22:%22variant_name%22}]},%22countryCode%22:%22US%22} HTTP/1.1
> Host: ophan.theguardian.com
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 204 No Content
< Cache-Control: no-cache, no-store
< Date: Thu, 05 Oct 2017 11:33:57 GMT
< Connection: keep-alive
<
* Connection #0 to host ophan.theguardian.com left intact
```

We also confirmed this event ended up in ElasticSearch

File: src/main/scala/com/gu/acquisition/services/DefaultOphanService.scala [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package com.gu.acquisition.services
 
 import java.io.IOException
+import java.net.URLEncoder
```
Comment (joelochlann) [inline src/main/scala/com/gu/acquisition/services/DefaultOphanService.scala]: I guess you can get rid of this one now?

**Review by joelochlann — APPROVED** [review]
Good catch Guy!

**Comment by Mullefa** [comment]
Cool, just tested with contributions front-end and everything looking good.

**Pull request merged**